controller/signUpController.js

```diff
@@ -0,0 +1,170 @@
+                const refreshToken = await jwt.sign({email: req.body.email},secretKey.token.key,{expiresIn: '30d'})
+                const user = new User({
+                    first_name : req.body.first_name,
+                    middle_name : req.body.nmiddle_name,
+                    last_name : req.body.last_name,    
+                    email: req.body.email,
+                    password: hash,
+                    phone : req.body.phone,
+                    dob : req.body.dob,
+                    address : req.body.address,
+                    user_role : role,
+                    refreshToken : refreshToken
+                })
+                user.save(async(err)=>{
+                    try{
+                        host=req.get('host');
+                        let transporter = nodemailer.createTransport({ service: 'gmail',
+                            port: 587, 
+                            secure :false,
+                            auth:{ user: process.env.GMAIL_USER, pass: process.env.GMAIL_PASSWORD } });
+                        let mailOptions = {  
+                            from : 'no-reply@mentors.com',
+                            to: user.email, 
+                            subject: 'Account Verification Token', 
+                            text: 'Hello,\n\n' + 'Please verify your account by clicking the link: \nhttp:\/\/' + host + '\/mentors\/confirmation/?token=' + token + '.\n' };
+                        await transporter.sendMail(mailOptions, async (err)=> {
+                            if (err) {
+                                return res.status(http.INTERNAL_SERVER_ERROR).json({ msg: http.getStatusText(http.INTERNAL_SERVER_ERROR) }); 
+                            }
+                        });
+                    }
+                    catch(err){
+                        next(err)
+                        res.json(err)
+                    }
+                    const response = {
+                        "status": "Success",
+                        "token": token,
+                        "refreshToken" :refreshToken,
+                        "data" : user,
+                        "message" : 'A verification email has been sent to ' + user.email + '.'
+                    }
+                    res.status(http.OK).json({response})
+                })
+            } 
+        } 
+    })
+    }    
+}
+
+exports.confirmation = (req,res)=>{
+    if (req.query.token) {
+        try {
+            decoded = jwt.verify(req.query.token, secretKey.token.key);
+        } 
+        catch (e) {
+            return res.status(http.UNAUTHORIZED).json({message:http.getStatusText(http.UNAUTHORIZED)});
+        }
+        const email = decoded.email;
+        User.findOne({ email: email },(err, user) => {
+            if (!user) return res.status(http.BAD_REQUEST).send({ status: 'Failure',msg: 'We were unable to find a user for this token.' });
```

Comment:
don't use hardcoded string they are hard to find. use proper name

Reply:
done